scripts/src/primazactl/primazamain/maincluster.py

```diff
@@ -1,38 +1,41 @@
 
-from primazactl.utils import primazaconfig
 from primazactl.utils import logger
 from primazactl.utils.kubeconfigwrapper import KubeConfigWrapper
-from primazactl.primazamain.constants import PRIMAZA_NAMESPACE
+from .constants import PRIMAZA_NAMESPACE, PRIMAZA_USER
 from primazactl.primaza.primazacluster import PrimazaCluster
-from primazactl.cmd.worker.create.constants import APPLICATION, SERVICE
 from primazactl.identity.kubeidentity import KubeIdentity
 from .clusterenvironment import ClusterEnvironment
+from primazactl.utils import names
 
 
 class MainCluster(PrimazaCluster):
     kubeconfig: KubeConfigWrapper = None
     kube_config_file: str
-
-    primaza_config: str | None = None
     primaza_version: str | None = None
 
     def __init__(
             self,
             cluster_name: str | None,
+            namespace: str | None,
             kubeconfig_path: str | None,
             config_file: str | None,
             version: str | None):
 
-        super().__init__(PRIMAZA_NAMESPACE,
-                         cluster_name,
-                         None,
-                         kubeconfig_path)
+        if not namespace:
+            namespace = PRIMAZA_NAMESPACE
```

Comment:
I'd propose to set it as required in `MainCluster`'s API and use the [Default](value used when an argument is not provided) parameter in argparse. WDYT?

Reply:
removed it, was missing from one location in the cmd structure. Of course if we ever make primazactl an api we will need it back.